evap/evaluation/tests/test_models.py

```diff
@@ -537,6 +538,21 @@ def test_textanswer_review_state(self):
             evaluation.TextAnswerReviewState.REVIEWED,
         )
 
+    def test_state_change_log_translated(self):
+        evaluation = baker.make(
+            Evaluation,
+            state=Evaluation.State.NEW,
+        )
+        evaluation.save()
+        with translation.override("de"):
+            # Refresh reference to evaluation, to force new log entry
+            evaluation = Evaluation.objects.get(pk=evaluation.pk)
+            evaluation.ready_for_editors()
+            evaluation.save()
+            log_entry = evaluation.related_logentries().order_by("id").last()
+            self.assertEqual(Evaluation.state_to_str(log_entry.data["state"]["change"][0]), "neu")
```

Comment:
(Semi-answering to https://github.com/e-valuation/EvaP/pull/2071#issuecomment-1819740914) The CI run currently fails here because "new" isn't translated in the test. You said that the translation for this should already be available, right? If so, this might be a bit harder to debug, feel free to ask me/us to look into what's going on here in that case.

Reply:
Yup, the translation entry you linked to should work here. I don't see why this fails from the code, will debug and get back to you.

Reply:
Turns out we've never used the hardcoded translation in any test so far. Tests only use translations stored in the database, such as english/german names of evaluations. We don't even compile the messages (`manage.py compilemessages`) on the CI runner, so the machine-readable translation file `django.mo` is missing on the CI runner. Thus, the translation entry linked above is not available in the test.

We could compile the message catalog on the test runner, but that would add a dependency between this test and the translations, and it would increase test run time.

Instead, I'd propose to just [patch](https://docs.python.org/3/library/unittest.mock.html#unittest.mock.patch) the `gettext` / `_` method for this test to "hardcode" the expected translation. Then, the test is independent of the actual translated string and works without the compiled messages available. You can do this by decorating the test as follows:

```python3
@patch("evap.evaluation.models._", new=(lambda string: "vorbereitet" if string == "prepared" else string))
def test_state_change_log_translated(self):
    ....
    # now only check for "prepared"/"vorbereitet", since we only included
    # that value in our mocked translation method
```
The same approach should also work with a view test.

Reply:
See [97983e8](https://github.com/e-valuation/EvaP/pull/2071/commits/97983e80e0dd1cc39808cc82cde64a0893ead3fa)